diff --git a/include/dirent.h b/include/dirent.h
index f195844..500eceb 100644
--- a/include/dirent.h
+++ b/include/dirent.h
@@ -364,6 +364,8 @@ static _WDIR *
 _wopendir(const wchar_t *dirname)
 {
 	wchar_t *p;
+	_WDIR *dirp;
+	DWORD n;
 
 	/* Must have directory name */
 	if (dirname == NULL || dirname[0] == '\0') {
@@ -372,7 +374,7 @@ _wopendir(const wchar_t *dirname)
 	}
 
 	/* Allocate new _WDIR structure */
-	_WDIR *dirp = (_WDIR*) malloc(sizeof(struct _WDIR));
+	dirp = (_WDIR*) malloc(sizeof(struct _WDIR));
 	if (!dirp)
 		return NULL;
 
@@ -390,7 +392,7 @@ _wopendir(const wchar_t *dirname)
 	 */
 #if !defined(WINAPI_FAMILY_PARTITION) || WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 	/* Desktop */
-	DWORD n = GetFullPathNameW(dirname, 0, NULL, NULL);
+	n = GetFullPathNameW(dirname, 0, NULL, NULL);
 #else
 	/* WinRT */
 	size_t n = wcslen(dirname);
@@ -479,6 +481,9 @@ static int
 _wreaddir_r(
 	_WDIR *dirp, struct _wdirent *entry, struct _wdirent **result)
 {
+	WIN32_FIND_DATAW *datap;
+	size_t i;
+	DWORD attr;
 	/* Validate directory handle */
 	if (!dirp || dirp->handle == INVALID_HANDLE_VALUE || !dirp->patt) {
 		dirent_set_errno(EBADF);
@@ -487,7 +492,7 @@ _wreaddir_r(
 	}
 
 	/* Read next directory entry */
-	WIN32_FIND_DATAW *datap = dirent_next(dirp);
+	datap = dirent_next(dirp);
 	if (!datap) {
 		/* Return NULL to indicate end of directory */
 		*result = NULL;
@@ -499,7 +504,7 @@ _wreaddir_r(
 	 * long to fit in to the destination buffer, then truncate file name
 	 * to PATH_MAX characters and zero-terminate the buffer.
 	 */
-	size_t i = 0;
+	i = 0;
 	while (i < PATH_MAX && datap->cFileName[i] != 0) {
 		entry->d_name[i] = datap->cFileName[i];
 		i++;
@@ -510,7 +515,7 @@ _wreaddir_r(
 	entry->d_namlen = i;
 
 	/* Determine file type */
-	DWORD attr = datap->dwFileAttributes;
+	attr = datap->dwFileAttributes;
 	if ((attr & FILE_ATTRIBUTE_DEVICE) != 0)
 		entry->d_type = DT_CHR;
 	else if ((attr & FILE_ATTRIBUTE_REPARSE_POINT) != 0)
@@ -597,6 +602,7 @@ static void _wrewinddir(_WDIR* dirp)
 static WIN32_FIND_DATAW *
 dirent_first(_WDIR *dirp)
 {
+	DWORD errorcode;
 	/* Open directory and retrieve the first entry */
 	dirp->handle = FindFirstFileExW(
 		dirp->patt, FindExInfoStandard, &dirp->data,
@@ -614,7 +620,7 @@ error:
 	dirp->invalid = 1;
 
 	/* Set error code */
-	DWORD errorcode = GetLastError();
+	errorcode = GetLastError();
 	switch (errorcode) {
 	case ERROR_ACCESS_DENIED:
 		/* No read access to directory */
@@ -681,6 +687,10 @@ dirent_hash(WIN32_FIND_DATAW *datap)
 /* Open directory stream using plain old C-string */
 static DIR *opendir(const char *dirname)
 {
+	struct DIR *dirp;
+	wchar_t wname[PATH_MAX + 1];
+	size_t n;
+	int error;
 	/* Must have directory name */
 	if (dirname == NULL || dirname[0] == '\0') {
 		dirent_set_errno(ENOENT);
@@ -688,14 +698,12 @@ static DIR *opendir(const char *dirname)
 	}
 
 	/* Allocate memory for DIR structure */
-	struct DIR *dirp = (DIR*) malloc(sizeof(struct DIR));
+	dirp = (DIR*) malloc(sizeof(struct DIR));
 	if (!dirp)
 		return NULL;
 
 	/* Convert directory name to wide-character string */
-	wchar_t wname[PATH_MAX + 1];
-	size_t n;
-	int error = mbstowcs_s(&n, wname, PATH_MAX + 1, dirname, PATH_MAX+1);
+	error = mbstowcs_s(&n, wname, PATH_MAX + 1, dirname, PATH_MAX+1);
 	if (error)
 		goto exit_failure;
 
@@ -738,6 +746,8 @@ static int
 readdir_r(
 	DIR *dirp, struct dirent *entry, struct dirent **result)
 {
+	size_t n;
+	int error;
 	/* Read next directory entry */
 	WIN32_FIND_DATAW *datap = dirent_next(dirp->wdirp);
 	if (!datap) {
@@ -747,8 +757,7 @@ readdir_r(
 	}
 
 	/* Attempt to convert file name to multi-byte string */
-	size_t n;
-	int error = wcstombs_s(
+	error = wcstombs_s(
 		&n, entry->d_name, PATH_MAX + 1,
 		datap->cFileName, PATH_MAX + 1);
 
@@ -768,11 +777,12 @@ readdir_r(
 	}
 
 	if (!error) {
+		DWORD attr;
 		/* Length of file name excluding zero terminator */
 		entry->d_namlen = n - 1;
 
 		/* Determine file type */
-		DWORD attr = datap->dwFileAttributes;
+		attr = datap->dwFileAttributes;
 		if ((attr & FILE_ATTRIBUTE_DEVICE) != 0)
 			entry->d_type = DT_CHR;
 		else if ((attr & FILE_ATTRIBUTE_REPARSE_POINT) != 0)
@@ -857,13 +867,14 @@ rewinddir(DIR *dirp)
 static long
 _wtelldir(_WDIR *dirp)
 {
+	WIN32_FIND_DATAW *datap;
 	if (!dirp || dirp->handle == INVALID_HANDLE_VALUE) {
 		dirent_set_errno(EBADF);
 		return /*failure*/-1;
 	}
 
 	/* Read next file entry */
-	WIN32_FIND_DATAW *datap = dirent_next(dirp);
+	datap = dirent_next(dirp);
 	if (!datap) {
 		/* End of directory stream */
 		return (long) ((~0UL) >> 1);
@@ -892,6 +903,7 @@ telldir(DIR *dirp)
 static void
 _wseekdir(_WDIR *dirp, long loc)
 {
+	long hash;
 	if (!dirp)
 		return;
 
@@ -918,7 +930,6 @@ _wseekdir(_WDIR *dirp, long loc)
 	 * with 100.00% accuracy! Moreover, the method is slow for large
 	 * directories.
 	 */
-	long hash;
 	do {
 		/* Read next directory entry */
 		WIN32_FIND_DATAW *datap = dirent_next(dirp);
@@ -965,6 +976,11 @@ scandir(
 	int (*filter)(const struct dirent*),
 	int (*compare)(const struct dirent**, const struct dirent**))
 {
+	struct dirent *tmp = NULL;
+	struct dirent **files = NULL;
+	size_t size = 0;
+	size_t allocated = 0;
+	size_t i;
 	int result;
 
 	/* Open directory stream */
@@ -975,11 +991,8 @@ scandir(
 	}
 
 	/* Read directory entries to memory */
-	struct dirent *tmp = NULL;
-	struct dirent **files = NULL;
-	size_t size = 0;
-	size_t allocated = 0;
 	while (1) {
+		struct dirent *entry;
 		/* Allocate room for a temporary directory entry */
 		if (!tmp) {
 			tmp = (struct dirent*) malloc(sizeof(struct dirent));
@@ -988,7 +1001,6 @@ scandir(
 		}
 
 		/* Read directory entry to temporary area */
-		struct dirent *entry;
 		if (readdir_r(dir, tmp, &entry) != /*OK*/0)
 			goto exit_failure;
 
@@ -1022,7 +1034,7 @@ scandir(
 
 exit_failure:
 	/* Release allocated entries */
-	for (size_t i = 0; i < size; i++) {
+	for (i = 0; i < size; i++) {
 		free(files[i]);
 	}
 
